File: ruby-rspec/example.spec.rb
RSpec.describe 'client' do

  require 'mailslurp_client'

  # configure the mailslurp client with an API Key
  MailSlurpClient.configure do |config|
    config.api_key['x-api-key'] = ENV['API_KEY']
  end

  it 'can create email addresses' do
    inbox_controller = MailSlurpClient::InboxControllerApi.new
    inbox = inbox_controller.create_inbox

    expect(inbox.id).not_to be_nil
    expect(inbox.email_address).to include("mailslurp.com")
  end

  it 'can send an email' do
    # create an inbox
    inbox_controller = MailSlurpClient::InboxControllerApi.new
    inbox = inbox_controller.create_inbox

    # send an email from the inbox (to the inbox's own address)
    inbox_controller.send_email(inbox.id, {
        send_email_options: {
            to: [inbox.email_address],
            subject: "Test",
            isHTML: true,
            body: <<-HEREDOC
              <h1>Hello!</h1>
              <p>MailSlurp supports HTML</p>
            HEREDOC
        }
    })
  end

  it 'can receive emails and extract content' do
    # create two inboxes
    inbox_controller = MailSlurpClient::InboxControllerApi.new
    inbox_1 = inbox_controller.create_inbox
    inbox_2 = inbox_controller.create_inbox

    # send an email from inbox 1 to inbox 2 (sends a real email)
    inbox_controller.send_email(inbox_1.id, {
        send_email_options: {
            to: [inbox_2.email_address],
            subject: "Test",
            body: "Your code is 123",
        }
    })

    # get emails from inbox2
    waitfor_controller = MailSlurpClient::WaitForControllerApi.new
    email = waitfor_controller.wait_for_latest_email({ inbox_id: inbox_2.id, unread_only: true })

    # verify email contents
    expect(email.subject).to include("Test")
    expect(email.body).to include("Your code is")
  end
end

File: javascript-axios/example.spec.js
/**
 * Example of how to create email addresses then send and receive emails
 * using Axios, NodeJs, and a MailSlurp free account.
 * See https://www.mailslurp.com/docs/
 */
const axios = require('axios');
const API_KEY = process.env.API_KEY;

describe("mailslurp axios example", () => {

  async function createInbox() {
    // make sure you have free API Key
    expect(API_KEY).toBeDefined();

    // call MailSlurp createInbox endpoint
    return await axios
      .post(`https://api.mailslurp.com/createInbox?apiKey=${API_KEY}`)
      .then(res => res.data);
  }

  test("create an inbox", async () => {
    const inbox = await createInbox();
    // expect an inbox to have been created
    expect(inbox.id).toBeDefined()
    expect(inbox.emailAddress).toBeDefined()
    expect(inbox.emailAddress).toMatch(/.+@mailslurp.com/)
  })

  test("can send a real email between two addresses", async () => {
    // create two inboxes
    const inbox1 = await createInbox();
    const inbox2 = await createInbox();

    // send email from inbox 1 to inbox 2
    // NOTE you can send emails to any address with MailSlurp
    await axios({
      method: 'post',
      url: `https://api.mailslurp.com/sendEmail?apiKey=${API_KEY}`,
      data: {
        senderId: inbox1.id,
        to: inbox2.emailAddress,
        subject: 'Hello inbox 2',
        body: 'Test from inbox 1'
      },
      json: true
    });

    // receive the email from inbox 2
    const email = await axios
      .get(`https://api.mailslurp.com/waitForLatestEmail?apiKey=${API_KEY}&inboxId=${inbox2.id}`)
      .then(res => res.data);
    expect(email.from).toEqual(inbox1.emailAddress);
    expect(email.subject).toEqual('Hello inbox 2');
    expect(email.body).toEqual('Test from inbox 1');
  })

})
